chasm/src/main/java/org/quiltmc/chasm/api/util/ClassInfo.java

```diff
@@ -0,0 +1,106 @@
+package org.quiltmc.chasm.api.util;
+
+import java.util.Arrays;
+
+import org.jetbrains.annotations.Contract;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Opcodes;
+
+/**
+ * Static information related to a specific class.
+ */
+public class ClassInfo {
+    public static final String OBJECT = "java/lang/Object";
+
+    private final String className;
+    private final String superClass;
+    private final String[] interfaces;
+    private final boolean isInterface;
+
+    /**
+     * Construct a new {@link ClassInfo}.
+     *
+     * @param className The name of the class (see {@link #getClassName()}).
+     * @param superClass The name of the super class (see {@link #getSuperClass()}).
+     * @param interfaces  The names of the implemented interfaces (see {@link #getInterfaces()}).
+     * @param isInterface  Whether this class is an interface (see {@link #isInterface}).
+     */
+    public ClassInfo(String className, String superClass, String[] interfaces, boolean isInterface) {
+        this.className = className;
+        this.superClass = superClass;
+        this.interfaces = interfaces;
+        this.isInterface = isInterface;
+    }
+
+    /**
+     * Returns the internal name of the class (See {@link org.objectweb.asm.Type#getInternalName}).
+     *
+     * @return The internal name of the class.
+     */
+    public String getClassName() {
+        return className;
+    }
+
+    /**
+     * Returns the internal name of the super class (See {@link org.objectweb.asm.Type#getInternalName}).
+     * For interfaces, this should return {@link #OBJECT}.
+     * Only for the {@link Object} class ({@link #OBJECT}) will this return null.
+     *
+     * @return The internal name of the super class.
+     */
+    @Contract(pure = true)
```

Comment:
Why is this contract annotation on this method but not on some of the others? Also, intellij is able to infer simple cases like this if the class is final (which there's no reason it shouldn't be). I find this annotation to be most useful on API interfaces.

Reply:
Because I copied the methods from the old Context. I'm removing it from ClassInfo.